compiler/src/iree/compiler/Preprocessing/Common/test/transpose_matmul.mlir

```diff
@@ -1,6 +1,9 @@
-// RUN: iree-opt --pass-pipeline="builtin.module(func.func(iree-preprocessing-transpose-matmul-pass{input=lhs}))" %s | FileCheck %s --check-prefixes=CHECK,LHS
-// RUN: iree-opt --pass-pipeline="builtin.module(func.func(iree-preprocessing-transpose-matmul-pass{input=rhs}))" %s | FileCheck %s --check-prefixes=CHECK,RHS
-// RUN: iree-opt --pass-pipeline="builtin.module(func.func(iree-preprocessing-transpose-matmul-pass))" %s | FileCheck %s --check-prefixes=CHECK,DISABLED
+// RUN: iree-compile --iree-preprocessing-pass-pipeline="builtin.module(util.func(iree-preprocessing-transpose-matmul-pass{input=lhs}))" \
+// RUN:   --iree-hal-target-backends=llvm-cpu --compile-to=preprocessing %s  | FileCheck %s --check-prefixes=CHECK,LHS
+// RUN: iree-compile --iree-preprocessing-pass-pipeline="builtin.module(util.func(iree-preprocessing-transpose-matmul-pass{input=rhs}))" \
+// RUN:   --iree-hal-target-backends=llvm-cpu --compile-to=preprocessing %s  | FileCheck %s --check-prefixes=CHECK,RHS
+// RUN: iree-compile --iree-preprocessing-pass-pipeline="builtin.module(util.func(iree-preprocessing-transpose-matmul-pass))" \
```

Comment:
The lit test as it was before was fine and `iree-compile` isn't preferred for lit tests like this anyway. I don't think there's a big rush to drop the flag because it's not causing any direct problems, just would be good to clean up. I think it's fine to wait on this until you've updated things downstream.

It would be good to document this better, but Preprocessing is intended to provide a plugin point for custom compiler passes/patterns. Currently that includes the `--preprocessing-pass-pipeline` flag like used here, transform dialect, PDL, and soon PDLL. The thinking is that this allows a fast path for shorter term modifications to the global pass pipeline without having to increase the flag surface in IREE core, but also without requiring downstream users to build C++ scaffolding adjacent to IREE just to add the passes they want. Although the mechanism for passing in the pass pipeline is admittedly unwieldy.

Reply:
SG, thanks for taking a look and for the context 🙏🏻  Let me just close this then.

> The lit test as it was before was fine and iree-compile isn't preferred for lit tests like this anyway. 

AFAIK, `--iree-preprocessing-pass-pipeline` is only exposed via `iree-compile`. Is there any other way to test preprocessing?

Reply:
preprocessing pass pipeline just uses the same logic for constructing a textual pipeline that `mlir-opt --pass-pipeline=...`, so testing the pass using `mlir/iree-opt` is good enough. There is a smoketest for the compiler flag here: https://github.com/iree-org/iree/blob/main/tests/compiler_driver/preprocessing_flags.mlir

Reply:
yeah, use iree-opt instead of iree-compile